src/util.cpp

```diff
@@ -621,15 +601,19 @@ const fs::path &GetDataDir(bool fNetSpecific)
             path = fs::system_complete(mapArgs["-datadir"]);
             if (!fs::is_directory(path))
             {
-                path = "";
+                // If the specified path is not a directory, then
+                // use the default path as that is the safest thing
+                // to do.
+                path = GetDefaultDataDir();
                 return path;
```

Comment:
Should probably log somewhere that the datadir argument isn't a valid path so users can tell why it isn't using the datadir they gave

Reply:
How are you going to do that when the logger itself can't initialize properly, since it uses the default data directory. This is an ugly early initialization issue.

Reply:
Hmm maybe it could print to the console instead of logging? It could also do the same datadir check after initializing the logger and log if it noticed that datadir was invalid, but that's a bit uglier.

Reply:
I think the best and most conservative thing to do is to throw a std::runtime_error and be done with it. This will stop execution early and prevent all sorts of mischief, and the error will be bubbled up to stdout/err. Committed the change.